### ISSUE TITLE
model.evaluate() and model.predict() have different accuracy results

### ISSUE THREAD

**Issue by dhwajr** [tflearn/tflearn#223]
model.evaluate(testX,testY) and model.predict(testX) give me different results. evaluate() prints validation_acc=70% but when I check the predicted values returned by model.predict, it has only 39% precision . Am I missing something?
result of model.evaluate():

``
 [0.62250000000000005]
``
result of classification_report on model.predict():

             precision    recall  f1-score   support

          0       0.39      0.51      0.44       195
          1       0.35      0.25      0.29       205
    avg / total       0.37      0.38      0.37       400

**Comment by dhwajr** [comment]
I have only two classes. ( 0 and 1 as shown above). model.evaluate gives me one number (0.6225) whereas model.predict gives me probability score for two classes for each test sample. I iterate through this test sample and assign a class which has higher probability to that sample. Then I checked those label results manually as well as using sklearn classification_report(), the precision and recall are different as shown above. (measured accuracy on labelled results =0.42).

Total samples to test are 400, batch size is 128 so it forms 4 batches.

Running it with only 1 sample and only 2 samples(each from 2 classes) has no issues and give me same output.

**Comment by aymericdamien** [comment]
That is really strange behavior.
I didn't notice any issue with my own implementation (based on lstm example).
```python
model.evaluate(): [0.52879999990463256]
model.predict(): 0.5288 (mean average over all samples)
```

I will try to check your code and data.

**Comment by dhwajr** [comment]
No issue in tflearn or tensorflow. Issuee was with the logic to assign a class if prob>0.5 . there were cases where both classes had exactly 0.5 prob which were causing diff in calculations.
